common/src/main/java/com/wynntils/features/user/ItemLockFeature.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.google.common.reflect.TypeToken;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigManager;
+import com.wynntils.core.config.TypeOverride;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.properties.FeatureInfo;
+import com.wynntils.core.features.properties.RegisterKeyBind;
+import com.wynntils.core.keybinds.KeyHolder;
+import com.wynntils.mc.event.ContainerClickEvent;
+import com.wynntils.mc.event.ContainerRenderEvent;
+import com.wynntils.mc.event.DropHeldItemEvent;
+import com.wynntils.mc.event.InventoryKeyPressEvent;
+import com.wynntils.mc.render.RenderUtils;
+import com.wynntils.mc.render.Texture;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wc.model.Character;
+import com.wynntils.wc.utils.WynnUtils;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.inventory.ClickType;
+import net.minecraft.world.inventory.Slot;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@FeatureInfo(category = "IWillRemoveThisWhenIFix#151")
+public class ItemLockFeature extends UserFeature {
+
+    @RegisterKeyBind
+    private final KeyHolder lockSlotKeybind = new KeyHolder(
+            "Lock Slot", GLFW.GLFW_KEY_H, "Wynntils", true, ItemLockFeature::tryChangeLockStateOnHoveredSlot);
+
+    @Config(visible = false)
+    private static Map<Integer, Set<Integer>> classSlotLockMap = new HashMap<>();
+
+    @TypeOverride
+    private static final Type classSlotLockMapType = new TypeToken<HashMap<Integer, Set<Integer>>>() {}.getType();
+
+    @Config
+    public static boolean blockAllActionsOnLockedItems = false;
+
+    @SubscribeEvent
+    public void onKeyPress(InventoryKeyPressEvent e) {
```

Comment:
Why is this not handled by registering our KeyHolder? Is there a special extra step for handling key presses in the inventory? Should we perhaps pass an additional boolean includeContainerKeypress to the KeyHolder constructor, to have it do this?

Reply:
I was not sure, itemscreenshot feature does the same "hack". Perhaps make an issue and deal with it later?

Reply:
Make an issue and deal with it later. It's a winning concept, pushing work to future you (or future us). :-) 

(Have you tested it it does not work without it?)

Reply:
It does not work.

Reply:
Tracked by #153.